chipsec/modules/common/cpu/spectre_v2.py

```diff
@@ -0,0 +1,267 @@
+
+
+    def is_supported(self):
+        return True
+
+    def check_spectre_mitigations( self ):
+
+        try:
+           cpu_thread_count = self.cs.msr.get_cpu_thread_count()
+        except:
+           cpu_thread_count = 1
+
+        #
+        # Read CPUID Leaf 07H
+        #
+        (r_eax, r_ebx, r_ecx, r_edx) = self.cs.cpu.cpuid( 0x7, 0x0 )
+        ibrs_ibpb_supported = (r_edx & chipsec.defines.BIT26) > 0
+        stibp_supported     = (r_edx & chipsec.defines.BIT27) > 0
+        arch_cap_supported  = (r_edx & chipsec.defines.BIT29) > 0
+        self.logger.log( "[*] CPUID.7H:EDX[26] = %d Indirect Branch Restricted Speculation (IBRS) & Predictor Barrier (IBPB)" % ibrs_ibpb_supported )
+        self.logger.log( "[*] CPUID.7H:EDX[27] = %d Single Thread Indirect Branch Predictors (STIBP)" % stibp_supported )
+        self.logger.log( "[*] CPUID.7H:EDX[29] = %d IA32_ARCH_CAPABILITIES" % arch_cap_supported )
+
+        if ibrs_ibpb_supported: self.logger.log_good( "CPU supports IBRS and IBPB" )
+        else:                   self.logger.log_bad( "CPU doesn't support IBRS and IBPB" )
+
+        if stibp_supported: self.logger.log_good( "CPU supports STIBP" )
+        else:               self.logger.log_bad( "CPU doesn't support STIBP" )
+
+        if not self.cs.is_register_defined( 'IA32_ARCH_CAPABILITIES' ) or \
+           not self.cs.is_register_defined( 'IA32_SPEC_CTRL' ):
+            self.logger.error( "couldn't find definition of required MSRs" )
+            return ModuleResult.ERROR
+
+
+        if arch_cap_supported:
+
+            ibrs_enh_supported = True
+            #rdcl_mitigation_supported = True
+            self.logger.log( "[*] checking enhanced IBRS support in IA32_ARCH_CAPABILITIES..." )
+            for tid in range(cpu_thread_count):
+                arch_cap_msr = 0
+                try:
+                    arch_cap_msr = self.cs.read_register( 'IA32_ARCH_CAPABILITIES', tid )
+                except chipsec.helper.oshelper.HWAccessViolationError:
+                    self.logger.error( "couldn't read IA32_ARCH_CAPABILITIES" )
+                    ibrs_enh_supported = False
+                    break
+
+                ibrs_all = self.cs.get_register_field( 'IA32_ARCH_CAPABILITIES', arch_cap_msr, 'IBRS_ALL' )
+                self.logger.log( "[*]   cpu%d: IBRS_ALL = %x" % (tid, ibrs_all) )
+                if 0 == ibrs_all:
+                    ibrs_enh_supported = False
+                    break
+
+                # @TODO: this checks for RDCL aka Meltdown (Variant 3) mitigation
+                #self.logger.log( "[*]   cpu%d: checking RDCL mitigation support..." % tid )
+                #rdcl_no = self.cs.get_register_field( 'IA32_ARCH_CAPABILITIES', arch_cap_msr, 'RDCL_NO' )
+                #self.logger.log( "[*]   cpu%d: RDCL_NO = %x" % (tid, rdcl_no) )
+                #if 0 == rdcl_no:
+                #    rdcl_mitigation_supported = False
+                #    break
```

Comment:
Do you want to remove the dead code for Meltdown and put it in its own module?  Same with the commented code in the block below.